strutil/strutil.go

```diff
@@ -367,6 +367,10 @@ func StrListDelete(s []string, d string) []string {
 
 	for index, element := range s {
 		if element == d {
+			// Using the provided slice as the basis for the return value can
+			// result in confusing behaviour, see https://go.dev/play/p/EAtNw4lLugu
+			c := make([]string, len(s))
+			copy(c, s)
```

Comment:
You don't actually use `c` here?
Should the next line actually be `return append(c[:index], c[index+1:]...)`?

Reply:
And instead of copying `s`, would it be better to build `c` from `s[:index]` and `s[index+1:]`?

Reply:
Haha, whoops, nice catch re `c` not being used.  I thought about being more clever in terms of indexing into `c`, but really I just wanted to fix the immediate problem, didn't want to get sucked into micro-optimization.

Reply:
I thought I responded to this yesterday, but apparently not.  I fixed the not-using-c issue (and the test I added failed without that).  I decided against the more clever approach you suggested, because I didn't want to pre-optimize, and because if I tried to make it optimal I probably wouldn't get around to fixing the bug before I got pulled off this for something else and then forgot about it.